docs/source/manual/validation.rst

```diff
@@ -0,0 +1,252 @@
+``@ValidationMethod`` to any ``boolean``-returning method which begins with ``is`` is a short and
+simple workaround:
+
+.. note::
+
+    Due to the rather daft JavaBeans conventions, when using ``@ValidationMethod``, the method must
+    begin with ``is`` (e.g., ``#isValidPortRange()``. This is a limitation of Hibernate Validator,
+    not Dropwizard.
+
+.. _man-validation-annotations-validated:
+
+``@Validated``
+**************
+
+The ``@Validated`` annotation behaves similar to ``@Valid``. The difference is that the
+``@Validated`` annotation allows for `validation groups`_ to be specifically set, instead of the
+default group. This is useful when different endpoints share the same entity but may have different
+requirements.
+
+.. _validation groups: https://docs.jboss.org/hibernate/validator/5.1/reference/en-US/html/chapter-groups.html
+
+Going back to our favorite ``Person`` class. Let's say we initially coded it such that ``name`` has
+to be non-empty, but realized that business requirements needs the max length to be no more than 5.
+Instead of blowing away our current version of our API and creating angry clients, we can accept
+both versions of the API but at different endpoints.
+
+.. code-block:: java
+
+    public interface Version1Checks { }
+
+    public interface Version2Checks { }
+
+    public class Person {
+        @NotEmpty(groups = Version1Checks.class)
+        @Length(max = 5, groups = Version2Checks.class)
+        private String name;
+
+        @JsonCreator
+        public Person(@JsonProperty("name") String name) {
+            this.name = name;
+        }
+
+        @JsonProperty
+        public String getName() {
+            return name;
+        }
+    }
+
+    @Path("/person")
+    @Produces(MediaType.APPLICATION_JSON)
+    public class PersonResource {
+        @POST
+        @Path("/v1")
+        public void createPersonV1(@Validated(Version1Checks.class) Person person) {
+        }
+
+        @POST
+        @Path("/v2")
+        public void createPersonV2(@Validated({Version1Checks.class, Version2Checks.class}) Person person) {
+        }
+    }
+
+Now, when clients hit ``/person/v1`` the ``Person`` entity will be checked by all the constraints
+that are a part of the ``Version1Checks`` group. If ``/person/v2`` is hit, then all the validations
+are performed.
+
+.. note::
+
+    Since interfaces can inherit other interfaces, ``Version2Checks`` can extend ``Version1Checks``
+    and wherever ``@Validated(Version2Checks.class)`` is used, version 1 constraints are checked
+    too.
+
+.. _man-validation-extending:
+
+Extending
+=========
+
+While Dropwizard provides good defaults for error messages, one size may not fit all and so there
+are a series of extension points. To register your own
+``ExceptionMapper<ConstraintViolationException>`` you'll need to first set
+``registerDefaultExceptionMappers`` to false in the configuration file or in code before registering
```

Comment:
`ConstraintViolationExceptionMapper` is a default exception mapper, and it will take precedence over a custom `ExceptionMapper<ConstraintViolationException>` that is registered.